manifests/pg_repack.pp

```diff
@@ -0,0 +1,137 @@
+# Maintenance pg_repack
+#
+# @summary Maintenance pg_repack
+
+class pe_databases::pg_repack (
+  Boolean $disable_maintenance = false,
+  Integer $jobs                = $facts['processors']['count'] / 4
+) {
+
+  $ensure_service = $disable_maintenance ? {
+    true  => stopped,
+    default => running,
+  }
+
+  $ensure_file = $disable_maintenance ? {
+    true    => absent,
+    default => present
+  }
+
+  # PE 2019.1 starting shipping versioned pe-postgres packages where all paths are versioned.
+  # So, prior to 2019.1 use a non-versioned path, and after use a versioned path.
+  # TODO: Use $pe_databases::psql_version after identifying why it is cast to ${psql_version}00000 in spec tests.
+  $postgresql_version = $facts['pe_postgresql_info']['installed_server_version']
+  $repack_executable = versioncmp('2019.1.0', $facts['pe_server_version']) ? {
+                          1       => '/opt/puppetlabs/server/apps/postgresql/bin/pg_repack',
+                          default => "/opt/puppetlabs/server/apps/postgresql/${postgresql_version}/bin/pg_repack"
+                          }
+
+  $repack          = "su - pe-postgres -s /bin/bash -c \"${repack_executable} -d pe-puppetdb"
```

Comment:
Is there any reason to continue switching to the pe-postgres user like this? Can't the service run as the pe-postgres user directly?

Reply:
We can and should, as it does work.  Thanks!